src/io/XTcpSocket.cpp

```diff
@@ -0,0 +1,124 @@
+#include "XTcpSocket.h"
+
+// system headers dependent
+
+#ifdef _MSC_VER
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#define SHUT_RDWR SD_BOTH
+#define MSG_NOSIGNAL 0
+#else
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+#endif
+
+// system headers independent
+#include <errno.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+typedef struct sockaddr sockaddr;
```

Comment:
In C++ it's generally good style to place type definition local to a .cpp file inside an unnamed namespace. Additionally, `using` would be more appropriate, :
```c++
namespace {
    using sockaddr = struct sockaddr;
}
```